src/main/java/com/algorithmia/algorithmHandler/ReflectionHelper.java

```diff
@@ -0,0 +1,56 @@
+package com.algorithmia.algorithmHandler;
+
+import java.io.Serializable;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class ReflectionHelper {
+
+
+    public static <T, R> String getMethodName(
+            DebuggableFunction<T, R> methodReference) {
+        SerializedLambda lambda = getLambda(methodReference);
+        return lambda.getImplMethodName();
+    }
+
+    public static <T1, T2, R> String getMethodName(DebuggableBifunction<T1, T2, R> methodReference) {
+        SerializedLambda lambda = getLambda(methodReference);
+        return lambda.getImplMethodName();
+    }
+
+    private static SerializedLambda getLambda(Serializable lambda) {
+        for (Class<?> cl = lambda.getClass(); cl != null; cl = cl.getSuperclass()) {
+            try {
+                Method m = cl.getDeclaredMethod("writeReplace");
+                m.setAccessible(true);
+                Object replacement = m.invoke(lambda);
+                if (!(replacement instanceof SerializedLambda)) {
+                    break; // custom interface implementation
+                }
+                SerializedLambda l = (SerializedLambda) replacement;
+                return l;
+            } catch (NoSuchMethodException e) {
+                // do nothing
```

Comment:
why is this a do nothing?  i think this whole method and class needs some javadoc/comments

Reply:
Agreed, I'll do my best to articulate what this is doing